lib/utils/timer.js

```diff
@@ -16,29 +16,81 @@ const process = require('process');
 
 // Simple timer making process.hrtime() easy to use
 class Timer {
+    // creates a timer. use start() to start measurements
     constructor() {
-        this.id = process.hrtime();
+        this._measured = false;
+        this._current = 0.0;
+        this._total = 0.0;
     }
 
-    end() {
-        if (!this.hrtimeElapsed) {
+    // starts or continues the timer
+    start() {
+        if (!this._id) {
+            this._id = process.hrtime();
+
+            this._measured = true;
+        }
+        return this;
+    }
+
+    // stops the timer if it was running and returns the total time in seconds as float
+    stop() {
+        if (this._id) {
             try {
-                this.hrtimeElapsed = process.hrtime(this.id);
+                this._current = hrtimeToSeconds(process.hrtime(this._id));
+                this._total += this._current;
```

Comment:
@alexkli Is the idea here that `this._id` will be used only to track `start()` & `stop()` (a.k.a `this._current` is coupled with `this_.id`) whereas `this._total` runs independent of `this._id` ?

Reply:
`this._id` is the identifier for the current [process.hrtime()](https://nodejs.org/api/process.html#process_process_hrtime_time) run. `stop()` ends it and the next `start()` would start a new one.

This allows to resume the timer, which we use for measuring N number of worker callbacks and get the sum automatically through this class. Previously that sum was calculated in multiple places in worker.js which just added lots of noise to the code there.